x/fees/keeper/keeper_test.go

```diff
@@ -0,0 +1,138 @@
+package keeper_test
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/fees/types"
+	posts "github.com/desmos-labs/desmos/x/posts/types"
+)
+
+func (suite *KeeperTestSuite) TestKeeper_SetParams() {
+	params := types.DefaultParams()
+	suite.keeper.SetParams(suite.ctx, params)
+
+	actualParams := suite.keeper.GetParams(suite.ctx)
+
+	suite.Equal(params, actualParams)
+}
+
+func (suite *KeeperTestSuite) TestKeeper_GetParams() {
+	params := types.DefaultParams()
+	suite.keeper.SetParams(suite.ctx, params)
+
+	actualParams := suite.keeper.GetParams(suite.ctx)
+
+	suite.Equal(params, actualParams)
+
+	tests := []struct {
+		name      string
+		params    *types.Params
+		expParams *types.Params
+	}{
+		{
+			name:      "Returning previously set params",
+			params:    &params,
+			expParams: &params,
+		},
+		{
+			name:      "Returning nothing",
+			params:    nil,
+			expParams: nil,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		suite.Run(test.name, func() {
+			if test.params != nil {
+				suite.keeper.SetParams(suite.ctx, *test.params)
+			}
+
+			if test.expParams != nil {
+				suite.Equal(*test.expParams, suite.keeper.GetParams(suite.ctx))
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestKeeper_CheckFees() {
+	// variables for later usage
+	timeZone, _ := time.LoadLocation("UTC")
+	pollData := posts.NewPollData(
+		"poll?",
+		time.Date(2050, 1, 1, 15, 15, 00, 000, timeZone),
+		posts.NewPollAnswers(
+			posts.NewPollAnswer("1", "Yes"),
+			posts.NewPollAnswer("2", "No"),
+		),
+		false,
+		true,
+	)
+	attachments := posts.NewAttachments(posts.NewAttachment("https://uri.com", "text/plain", nil))
+	id := "dd065b70feb810a8c6f535cf670fe6e3534085221fa964ed2660ebca93f910d1"
+
+	var testOwner = "cosmos1cjf97gpzwmaf30pzvaargfgr884mpp5ak8f7ns"
+
+	tests := []struct {
+		name      string
+		params    types.Params
+		givenFees sdk.Coins
+		msgs      []sdk.Msg
+		expError  error
```

Comment:
Instead of specifying an error, I suggest just checking if an error is raised or not. It has no particular advantage for us to check the specific error message and type. Instead, it may make it harder to maintain tests in the future. For example, if we change the error message, we need to also update the test. This is a huge overhead to the development process, without any particular advantage for us IMO.

This means making `expError bool` instead of `error`, and then inside the test simply do: 


```golang
if test.expError {
  suite.Require().Error(err)
} else {
  suite.Require.NoError(err)
}
```